Microsoft.Toolkit.Uwp.UI.Controls/Eyedropper/Eyedropper.cs

```diff
@@ -172,49 +240,65 @@ private void TargetGrid_PointerEntered(object sender, PointerRoutedEventArgs e)
 
         private async void Eyedropper_DpiChanged(DisplayInformation sender, object args)
         {
+            _currentDpi = sender.LogicalDpi;
             await UpdateAppScreenshotAsync();
         }
 
         private async void TargetGrid_PointerReleased(object sender, PointerRoutedEventArgs e)
         {
-            var pointer = e.Pointer;
-            if (pointer.PointerId == _pointerId)
+            var point = e.GetCurrentPoint(_rootGrid);
+            await InternalPointerReleasedAsync(e.Pointer.PointerId, point.Position);
+        }
+
+        internal async Task InternalPointerReleasedAsync(uint pointerId, Point position)
+        {
+            if (pointerId == _pointerId)
             {
-                var point = e.GetCurrentPoint(_rootGrid);
                 if (_appScreenshot == null)
                 {
                     await UpdateAppScreenshotAsync();
                 }
 
-                UpdateEyedropper(point.Position);
-                PickCompleted?.Invoke(this, EventArgs.Empty);
+                UpdateEyedropper(position);
                 _pointerId = null;
-                if (!_taskSource.Task.IsCanceled)
+                if (_taskSource != null && !_taskSource.Task.IsCanceled)
                 {
-                    _taskSource.SetResult(Color);
+                    _taskSource.TrySetResult(Color);
                 }
+
+                PickCompleted?.Invoke(this, EventArgs.Empty);
             }
         }
 
         private void TargetGrid_PointerMoved(object sender, PointerRoutedEventArgs e)
         {
             var pointer = e.Pointer;
-            if (pointer.PointerId == _pointerId)
+            var point = e.GetCurrentPoint(_rootGrid);
+            InternalPointerMoved(pointer.PointerId, point.Position);
+        }
+
+        internal void InternalPointerMoved(uint pointerId, Point position)
+        {
+            if (pointerId == _pointerId)
             {
-                var point = e.GetCurrentPoint(_rootGrid);
-                UpdateEyedropper(point.Position);
+                UpdateEyedropper(position);
             }
         }
 
         private async void TargetGrid_PointerPressed(object sender, PointerRoutedEventArgs e)
         {
-            _pointerId = e.Pointer.PointerId;
             var point = e.GetCurrentPoint(_rootGrid);
+            await InternalPointerPressedAsync(e.Pointer.PointerId, point.Position, e.Pointer.PointerDeviceType);
+        }
+
+        internal async Task InternalPointerPressedAsync(uint pointerId, Point position, Windows.Devices.Input.PointerDeviceType pointerDeviceType)
```

Comment:
Did you want to add comments that these abstractions were for testing purposes?

Reply:
Done.